test/augmentation/apply/test_tf_applier.py

```diff
@@ -94,6 +99,53 @@ def test_tf_applier_keep_original_modify_in_place(self):
         )
         self.assertTrue(all(x.d == y for x, y in zip(data, get_data_dict())))
 
+    def test_tf_applier_generator(self) -> None:
+        data = self._get_x_namespace()
+        applier = TFApplier([square], policy, k=1, keep_original=False)
+        batches_expected = [[1, 16], [81]]
+        gen = applier.apply_generator(data, batch_size=2)
+        for batch, batch_expected in zip(gen, batches_expected):
+            self.assertEqual([x.num for x in batch], batch_expected)
+        self.assertEqual([x.num for x in data], DATA)
+
+    def test_tf_applier_multi_generator(self) -> None:
+        data = self._get_x_namespace()
+        applier = TFApplier([square], policy, k=2, keep_original=False)
+        batches_expected = [[1, 1, 16, 16], [81, 81]]
+        gen = applier.apply_generator(data, batch_size=2)
+        for batch, batch_expected in zip(gen, batches_expected):
+            self.assertEqual([x.num for x in batch], batch_expected)
+        self.assertEqual([x.num for x in data], DATA)
+
+    def test_tf_applier_keep_original_generator(self) -> None:
+        data = self._get_x_namespace()
+        applier = TFApplier([square], policy, k=2, keep_original=True)
+        batches_expected = [[1, 1, 1, 2, 16, 16], [3, 81, 81]]
+        gen = applier.apply_generator(data, batch_size=2)
+        for batch, batch_expected in zip(gen, batches_expected):
+            self.assertEqual([x.num for x in batch], batch_expected)
+        self.assertEqual([x.num for x in data], DATA)
+
+    def test_tf_applier_returns_none_generator(self) -> None:
+        data = self._get_x_namespace()
+        applier = TFApplier([square_returns_none], policy, k=2, keep_original=True)
+        batches_expected = [[1, 1, 1, 2], [3, 81, 81]]
+        gen = applier.apply_generator(data, batch_size=2)
+        for batch, batch_expected in zip(gen, batches_expected):
+            self.assertEqual([x.num for x in batch], batch_expected)
+        self.assertEqual([x.num for x in data], DATA)
+
+    def test_tf_applier_keep_original_modify_in_place_generator(self) -> None:
```

Comment:
It's a little hard to follow what this test is doing, both since "modify in place _and_ keep original" is a bit of a paradox, and because both the assert test below and the DATA_IN_PLACE_EXPECTED are generated programmatically, so for a user to understand the test they need to hold a few things in their heads at once.

Reply:
Ok sg